File: IS_Lab2/IS_Lab2_JSON/Kod źródłowy/deserialize_json.py
# -*- coding: utf-8 -*-
"""
deserialize json
"""
import json


class DeserializeJson:
    # konstruktor
    def __init__(self, filename):
        print("let's deserialize something")
        tempdata = open(filename, encoding="utf8")
        self.data = json.load(tempdata)

    # przykładowe statystyki
    def somestats(self):
        example_stat = 0
        dictionary = dict()
        for dep in self.data:
            if dep['typ_JST'] == 'GM' and dep['Województwo'] == 'dolnośląskie':
                example_stat += 1

            wojewodztwo = dep['Województwo'].replace(" ", "")
            if wojewodztwo not in dictionary:
                dictionary[wojewodztwo] = dict()

            if dep['typ_JST'] not in dictionary[wojewodztwo]:
                dictionary[wojewodztwo][dep['typ_JST']] = 0
            dictionary[wojewodztwo][dep['typ_JST']] += 1

        print('liczba urzędów miejskich w województwie dolnośląskim: ' + ' ' + str(example_stat))

        for records in dictionary:
            print("Województwo: " + str(records))
            for record in dictionary[records]:
                print("Liczba wystąpień urzędów typu " + str(record) + ": " + str(dictionary[records][record]))
            print()

File: IS_Lab2/IS_Lab2_C#/Kod źródłowy/Deserialize_json.cs
using Newtonsoft.Json;

namespace Lab2
{
    internal class Deserialize_json
    {
        public List<Dictionary<string, dynamic>> data = new List<Dictionary<string, dynamic>>();

        public Deserialize_json(string filename)
        {
            Console.WriteLine("let's deserialize something");
            using (StreamReader sr = new StreamReader(filename))
            {
                string json = sr.ReadToEnd();
                data = JsonConvert.DeserializeObject<List<Dictionary<string, dynamic>>>(json);
            }
            Console.WriteLine("it is done");
        }

        public void Somestats()
        {
            var example_stat = 0;
            var dictionary = new Dictionary<string, Dictionary<string, int>>();

            foreach (var record in data)
            {
                if (record["typ_JST"] == "GM" && record["Województwo"] == "dolnośląskie")
                    example_stat++;


                var wojewodztwo = record["Województwo"].Replace(" ", "");

                if (!dictionary.ContainsKey(wojewodztwo))
                {
                    var dict = new Dictionary<string, int>();
                    dictionary[wojewodztwo] = dict;
                }

                if (!dictionary[wojewodztwo].ContainsKey(record["typ_JST"]))
                {
                    dictionary[wojewodztwo][record["typ_JST"]] = 0;
                }
                dictionary[wojewodztwo][record["typ_JST"]]++;
            }

            Console.WriteLine("Liczba urzędów miejskich w województwie dolnośląskim: " + ' ' + example_stat);
            Console.WriteLine();

            foreach (var dict in dictionary)
            {
                Console.WriteLine("Województwo: " + dict.Key);
                foreach (var record in dict.Value)
                {
                    Console.WriteLine("Liczba wystąpień urzędów typu " + record.Key + ": " + record.Value);
                }
                Console.WriteLine();
            }
        }
    }
}
